src/integ_test_resources/ios/sdk/integration/cdk/custom_resources/iot_custom_authorizerUserPass_provider/iot_custom_authorizerUserPass_provider.py

```diff
@@ -0,0 +1,107 @@
+import boto3
+
+# Note: Make this dynamic if there is more than one custom authorizer in the stack
+physical_id = "iot_custom_authorizerUserPass_provider_20200508074521"
+domain_configuration_name = "aws_test_iot_custom_authorizerUserPass"
+default_authorizer_name = "iot_custom_authorizerUserPass"
+
+def on_event(event, __):
+    print(f"### on_event({event})")
+    request_type = event["RequestType"]
+    if request_type == "Create":
+        return on_create_or_update(event)
+    if request_type == "Update":
+        return on_create_or_update(event)
+    if request_type == "Delete":
+        return on_delete(event)
+    raise Exception("Invalid request type: %s" % request_type)
+
+
+def on_create_or_update(event):
+    print(f"### on_create_or_update({event})")
+    custom_authorizer_function_arn = event["ResourceProperties"]["authorizer_function_arn"]
```

Comment:
Since you use `event["ResourceProperties"]` so many times, you could `props = event["ResourceProperties"]` it, perhaps. At that point, `props["authorizer_function_arn"]` is so short that you hardly even need all of these cache variables. You could just make direct inline reference to the `props` dict.

Reply:
I'm going to leave this as is.. this was largely a copy and paste from the original iot_custom_authorizer_provider, and i see more value in keeping the two pieces of code similar.